### PR TITLE
chore(whisper): Pass through variables down to whisper

### PR DESCRIPTION
**Pull request by CollectiveUnicorn**
Ensures that all variables are passed down to the actual whisper model
* `language, temperature, prompt`

### CODE REVIEW

**Review by justinthelaw — DISMISSED** [review]

**Review by gphorvath — DISMISSED** [review]

File: packages/whisper/main.py [hunk 1]
```diff
@@ -18,10 +18,21 @@ def make_transcribe_request(filename, task, language, temperature, prompt):
     device = "cuda" if GPU_ENABLED else "cpu"
     model = WhisperModel(model_path, device=device, compute_type="float32")
 
-    segments, info = model.transcribe(filename, task=task, beam_size=5)
+    # Prepare kwargs with non-None values
+    kwargs = {}
+    if task is not None:
+        kwargs["task"] = task
```
Comment (gphorvath) [inline packages/whisper/main.py:24]: should we validate that task is `transcribe` or `translate` or just leave that up to faster-whisper?

File: packages/whisper/main.py
Comment (gphorvath) [inline packages/whisper/main.py:1]: Is this a good time to upgrade the version of faster-whisper? I noticed we are on `0.1.0` which is from November 2023. Version 1.0.3 (July 2024) seem to work locally for me with these changes.

File: packages/whisper/main.py [hunk 2]
```diff
@@ -18,10 +18,21 @@ def make_transcribe_request(filename, task, language, temperature, prompt):
...
+    if language is not None:
+        kwargs["language"] = language
```
Comment (gphorvath) [inline packages/whisper/main.py:26]: faster-whisper has `supported_languages() -> List[str]` that we could use to validate an incoming language

File: packages/whisper/main.py
Comment (CollectiveUnicorn) [reply packages/whisper/main.py:1]: Yep, bumping the version

File: packages/whisper/main.py (hunk 2 above)
Comment (CollectiveUnicorn) [reply packages/whisper/main.py]: Added a check for this and made it default to automatically recognizing the language if that fails

File: packages/whisper/main.py (hunk 1 above)
Comment (CollectiveUnicorn) [reply packages/whisper/main.py]: I've added checks here now

**Review by justinthelaw — APPROVED** [review]

**Pull request merged**